agent_build/docker/Dockerfile.base

```diff
@@ -0,0 +1,81 @@
+# Base image that creates all necessary dependencies for the scalyr-agent docker image.
+# NOTE: multi-stage builds require Docker 17.05 or greater
+
+# Suffix for the python dockerhub image. For now can be:
+#   - 'slim' for buster based image
+#   - 'alpine' for alpine based image.
+ARG BASE_IMAGE_SUFFIX
+
+# Install dependency packages for buster.
+FROM python:3.8.10-slim as scalyr-dependencies-slim
+MAINTAINER Scalyr Inc <support@scalyr.com>
+
+RUN apt-get update && apt-get install -y build-essential git tar curl
+
+
+# Install dependency packages for alpine.
+FROM python:3.8.10-alpine as scalyr-dependencies-alpine
+MAINTAINER Scalyr Inc <support@scalyr.com>
+
+RUN apk update && apk add --virtual build-dependencies \
+    binutils \
+    build-base \
+    gcc \
+    g++ \
+    make \
+    curl \
+    python3-dev \
+    patchelf \
+    git
+
+# Install rust and cargo. It is needed to build some of the Python dependencies of the agent.
+FROM scalyr-dependencies-$BASE_IMAGE_SUFFIX as scalyr-dependencies-rust-
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```

Comment:
I just did curl + bash as a quick solution to test it out a while ago, but to be on the safe side we should probably either vendor the installer script file inside this repo or download the installer file and check the checksum against the one statically specified in this Dockerfile.

Reply:
Btw - does building wheel works on armv7 under alpine? I had issues and couldn't get it to work since rust toolchain for cross compiling is not available for armv7 + musl.

Reply:
EDIT: Never mind, I see this comment below.

>  # if TARGETVARIANT is 'v7', then we use stage without rust.

IIRC, that is only the case for armv7 + alpine (musl) and not armv7 + buster (glibc).